src/main/java/org/jitsi/videobridge/EndpointMessageTransport.java

```diff
@@ -535,4 +537,69 @@ public BridgeChannelMessage lastN(LastNMessage message)
         return null;
     }
 
+    /**
+     * Handles an opaque message from this {@code Endpoint} that should be forwarded to either: a) another client in
+     * this conference (1:1 message) or b) all other clients in this conference (broadcast message).
+     *
+     * @param message the message that was received from the endpoint.
+     *
+     * EndpointMessage definition:
+     * 'to': If the 'to' field contains the endpoint id of another endpoint in this conference, the message will be
+     * treated as a 1:1 message and forwarded just to that endpoint. If the 'to' field is an empty string, the message
+     * will be treated as a broadcast and sent to all other endpoints in this conference.
+     * 'msgPayload': An opaque payload. The bridge does not need to know or care what is contained in the 'msgPayload'
+     * field, it will just forward it blindly.
+     */
+    @Override
+    public BridgeChannelMessage endpointMessage(EndpointMessage message)
+    {
+        String to = message.getTo();
+
+        // First insert/overwrite the "from" to prevent spoofing.
+        String from = endpoint.getID();
+        message.setFrom(from);
+
+        Conference conference = endpoint.getConference();
+
+        if (conference == null || conference.isExpired())
+        {
+            logger.warn("Unable to send EndpointMessage, conference is null or expired");
+            return null;
+        }
+
+        boolean sendToOcto;
+
+        List<AbstractEndpoint> targets;
+        if ("".equals(to))
```

Comment:
nit: use an `isEmpty` helper ?

Reply:
I added a helper in EndpointMessage itself